src/codegen/compile.ml

```diff
@@ -7784,16 +8006,86 @@ and compile_exp (env : E.t) ae exp =
     | OtherPrim ("arrayToBlob"|"arrayMutToBlob"), e ->
       const_sr SR.Vanilla (Arr.toBlob env)
 
-    | OtherPrim ("stableMemoryLoadNat32"), [e] ->
+    | OtherPrim ("stableMemoryLoadNat32"|"stableMemoryLoadInt32"), [e] ->
       SR.UnboxedWord32,
       compile_exp_as env ae SR.UnboxedWord32 e ^^
-      StableMem.load_nat32 env
+      StableMem.load_word32 env
 
-    | OtherPrim ("stableMemoryStoreNat32"), [e1; e2] ->
+    | OtherPrim ("stableMemoryStoreNat32"|"stableMemoryStoreInt32"), [e1; e2] ->
       SR.unit,
       compile_exp_as env ae SR.UnboxedWord32 e1 ^^
       compile_exp_as env ae SR.UnboxedWord32 e2 ^^
-      StableMem.store_nat32 env
+      StableMem.store_word32 env
+
+    | OtherPrim ("stableMemoryLoadNat8"), [e] ->
+      SR.Vanilla,
+      compile_exp_as env ae SR.UnboxedWord32 e ^^
+      StableMem.load_word8 env ^^
+      TaggedSmallWord.msb_adjust Type.Nat8
+
+    | OtherPrim ("stableMemoryLoadInt8"), [e] ->
+      SR.Vanilla,
+      compile_exp_as env ae SR.UnboxedWord32 e ^^
+      StableMem.load_word8 env ^^
+      TaggedSmallWord.msb_adjust Type.Int8
+
+    | OtherPrim ("stableMemoryStoreNat8"), [e1; e2] ->
+      SR.unit,
+      compile_exp_as env ae SR.UnboxedWord32 e1 ^^
+      compile_exp_as env ae SR.Vanilla e2 ^^ TaggedSmallWord.lsb_adjust Type.Nat8 ^^
+      StableMem.store_word8 env
```

Comment:
I wonder if you could sink the `m/lsb_adjusts` into the `StableMem.store_word8`.

Reply:
Then some of these could be collapsed using `or` patterns.

Reply:
Yeah, I'm not sure Its worth it . I don't want to retest the string lit and don't want to rely on the type of e2 (that could, in principal, be `Non`). I'll just leave it, but thanks for the suggestion.